### ISSUE TITLE
Fail gracefully when connection cannot be established

### ISSUE THREAD

**Issue by gajus** [ts-safeql/safeql#349]
It seems like, if a connection to the database cannot be established, safeql will re-attempt the connection for every file. This can lead to ESLint effectively hanging indefinitely (which, in case of IDE/auto-fix integrations, makes it impossible to save changes).

In an ideal scenario, if a connection cannot be established, the checker would timeout after some time and remember that it was unable to connect with some sort of debounce-based logic, i.e. so if it failed to connect once, it doesn't retry on that run, and just skips over the rule.

This is important in a local development env where the associated database might be rebuilt/unavailable frequently, e.g. In our case, any time a migration change is made/branch is switched, the database is rebuilt. Rebuilding might take a minute. At the moment, due to the way that ESLint/Safeql/VSCode interwork, this means that one cannot basically save changes (b/c ESLint keeps timing out).

Some sort of option for local development that would make this failure more graceful would majorly improve the DX.

**Comment by Newbie012** [comment]
Makes sense. I could think of two possibilities. Either fail the rule entirely or do some sort of retries with exponential backoff. Thoughts?

**Comment by gajus** [comment]
> Makes sense. I could think of two possibilities. Either fail the rule entirely or do some sort of retries with exponential backoff. Thoughts?

Between these two options, I would just start failing.

I'd rather become quickly aware that it is failing than sit wondering what's happening.

This is esp. important in the context of VSCode auto-correct – because it prevents you from saving file changes while a rule is pending.

**Comment by Newbie012** [comment]
I see. I'll address it shortly.
